app/main/controller/image_similarity_controller.py

```diff
@@ -0,0 +1,63 @@
+import json
+from flask import request, current_app as app
+from flask_restplus import Resource, Namespace, fields
+from app.main import db
+from app.main.model.image import ImageModel
+from app.main.lib.fields import JsonObject
+from sqlalchemy import text
+
+api = Namespace('image_similarity', description='image similarity operations')
+image_similarity_request = api.model('image_similarity_request', {
+  'url': fields.String(required=True, description='image URL to be stored or queried for similarity'),
+  'threshold': fields.Integer(required=False, default=1, description='minimum score to consider, between 0 and 64 (defaults to 1)'),
+  'context': JsonObject(required=False, description='context')
+})
+
+@api.route('/')
+class ImageSimilarityResource(Resource):
+  @api.response(200, 'image signature successfully stored in the similarity database.')
+  @api.doc('Store an image signature in the similarity database')
+  @api.expect(image_similarity_request, validate=True)
+  def post(self):
+    result = True
+    image = ImageModel.from_url(request.json['url'], request.json['context'])
+    db.session.add(image)
+    db.session.commit()
```

Comment:
I noticed that when an identical image is sent, an exception is thrown because the SHA 256 already exists in the database. The most important thing is that we make sure that duplicate hashes are not stored, which is already happening... but would be good to handle this case gracefully.

Reply:
I am actually not sure we should NOT store duplicate images... Consider the case of the same uploaded image to two different workspaces. We should either: a) store two records, each with its own context, or b) support multi-team contexts.

Reply:
or (c) have our own "unique key" which could be, on this case, hash + team id

Reply:
We already have a serial id, so can we relax the SHA256 uniqueness for now?

Reply:
I think so... on Check API side, for example, we allow two identical images under the same team (we just de-duplicate on Smooch Bot).